### DISCUSSION TITLE
Set locale for dates formatting globally

### DISCUSSION THREAD

**Discussion by cjoecker** [sergiodxa/remix-i18next#224] [Q&A] [answered]
`date-fns` has the function `setDefaultOptions` to set the locale globally. This safe the effort for passing the locale to every `format` function in order to get the date in the users locale.

Is there a way to use the `setDefaultOptions` using the `i18next` locale so that there are no hydration errors if the server has a different locale than the client?

**Comment by sergiodxa** [comment, marked as answer]
You could do it in your loaders, but note that unless you're 100% sure every request will create a new instance of your server app, your server may be shared between users with different locales, so setting something like this globally will most likely cause issues.

I recommend you to set it on every place you need it or use something like `AsyncLocalStorage` to make it available to nested code.
